src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -278,6 +248,19 @@ internal static void FreeSslContext(IntPtr sslContextPtr)
 
         #region private methods
 
+        private static void FreeBio(SslContext sslContext)
+        {
+            if (IntPtr.Zero != sslContext.readBioPtr)
+            {
+                Interop.libcrypto.BIO_free(sslContext.readBioPtr);
+            }
+
+            if (IntPtr.Zero != sslContext.writeBioPtr)
+            {
+                Interop.libcrypto.BIO_free(sslContext.writeBioPtr);
```

Comment:
This is an example of where having the SslContext as a class instead of a struct would help.  Here you could set readBioPtr and writeBioPtr to null, preventing future accidental double-frees.